### PR TITLE
fix width detection of array querying function in case and case item expressions

### PR DESCRIPTION
**Pull request by zachjs**
I also removed the unnecessary shadowing of `width_hint` and `sign_hint` in the corresponding case in `simplify()`.

This fixes the issue raised in #3082, while also handling the (now covered) edge case that requires the existing ordering of `detectSignWidth` and `simplify`.

### CODE REVIEW

File: tests/simple/case_expr_query.sv [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    case ($low  (out)) 0:
+    case ($left (out)) 5:
+    case ($right(out)) 0:
+    case (6) $bits (out):
+    case (6) $size (out):
+    case (5) $high (out):
+    case (0) $low  (out):
+    case (5) $left (out):
+    case (0) $right(out):
+        out = '1;
```
Comment (clairexen) [inline tests/simple/case_expr_query.sv:17]: should `out` be initialized to `'0` at the start of this `always_comb` block?

**Review by clairexen — APPROVED** [review]
lgtm

File: tests/simple/case_expr_query.sv (hunk 1 above)
Comment (zachjs) [reply tests/simple/case_expr_query.sv:17]: I don't think it's strictly necessary, but I'll make that change!
